src/app/api/posts/route.ts

```diff
@@ -7,16 +7,47 @@ export async function GET(req: Request) {
   try {
     const regex = new RegExp(search, 'ig');
 
-    return new Response(
-      JSON.stringify(
-        posts.filter((post) => {
-          return (
-            regex.test(post.full) ||
-            post.full.toLowerCase().includes(search.toLowerCase())
-          );
-        })
-      )
-    );
+    const filteredPosts = posts
+      .filter((post) => {
+        return (
+          regex.test(post.full) ||
+          post.full.toLowerCase().includes(search.toLowerCase())
+        );
+      })
+      // update description
+      .map((post) => {
+        if (!search) return post;
+        const fullTextWithoutTitle = post.full.replace(post.title, '');
+        if (!fullTextWithoutTitle.toLowerCase().includes(search.toLowerCase()))
+          return post;
```

Comment:
Na linha 21 ele retornaria o post normal caso não incluisse a pesquisa literalmente, o lance é que a pesquisa tbm aceita regex e nesse caso ia falhar já que tecnicamente ela não "inclui" tipo um [a] literal.
Acho que uma solução mais bonitinha é criar uma função reutilizável ali em cima que cheque ambos os métodos.

```suggestion
    const includesSearch = (text: string): boolean => {
      return (
        regex.test(text) ||
        text.toLowerCase().includes(search.toLowerCase())
      );
    }
    
    const filteredPosts = posts
      .filter((post) => includesSearch(post.full))
      // update description
      .map((post) => {
        if (!search) return post;
        const fullTextWithoutTitle = post.full.replace(post.title, '');
        if (!includesSearch(fullTextWithoutTitle)) return post
```